e2e/cypress/support/commands.js

```diff
@@ -134,3 +136,165 @@ Cypress.Commands.add('deleteMetadata', (metadataTrackName) => {
+    cy.wrap($sample).find('.anticon-edit').click();
+    log.snapshot('editing-sample-name');
+
+    cy.wrap($sample).find('input').type('{selectall}{backspace}').type(newName || randomTestName);
+    log.snapshot('edited-sample-name');
+
+    cy.wrap($sample).find('.anticon-check').click();
+    log.snapshot('save-new-sample-name');
+  });
+
+  log.end();
+});
+
+Cypress.Commands.add('launchAnalysis', () => {
+  const log = Cypress.log({
+    displayName: 'Launching analysis',
+    message: ['launch analysis'],
+    autoEnd: false,
+  });
+
+  cy.get('[data-test-id="launch-analysis-button"]').click();
+  log.snapshot('launch-analysis');
+
+  cy.get('[data-test-class="launch-analysis-item"]').contains('button', 'Launch').first().click();
+  log.snapshot('launch-experiment');
+  log.end();
+});
+
+/** Valid values are text in the links in the navigation menu */
+Cypress.Commands.add('navigateTo', (page, config = {}) => {
+  Cypress.log({
+    displayName: `Navigate using to ${page}`,
+    message: [`navigate to ${page}`],
+  });
+
+  cy.get('[data-test-id="navigation-menu"]').contains('a', page).click(config);
+});
+
+Cypress.Commands.add('listenOnWebsocket', (fn) => {
+  const webSocketUrl = Cypress.env('webSocketUrl');
+
+  Cypress.log({
+    displayName: 'Connect to websocket',
+    message: `Connect to websocket on ${webSocketUrl}`,
+  });
+
+  const io = socketIOClient(webSocketUrl, { transports: ['websocket'] });
+
+  // Callbacks will have scope to io object
+  fn(io);
+});
+
+Cypress.Commands.add('waitForGem2s', (experimentId, config = {}) => {
+  Cypress.log({
+    displayName: 'GEM2S',
+    message: 'Waiting for GEM2S to complete',
+  });
+
+  const numGem2sSteps = 7;
+  const gem2sStepTimeOut = (60 * 1000) * 5; // 5 minutes;
+
+  cy.listenOnWebsocket((socket) => {
+    const gem2sResponses = [];
+
+    socket.on(`ExperimentUpdates-${experimentId}`, (update) => {
+      gem2sResponses.push(update);
+    });
+
+    // Create array [0, 1, 2, ... numGem2sSteps]
+    const waitForNSteps = [...Array(numGem2sSteps).keys()];
+
+    waitForNSteps.forEach((step) => {
+      cy.waitUntil(() => {
+        if (gem2sResponses.length === 0) return false;
+        const latestResponse = gem2sResponses.pop();
+        return latestResponse;
+      },
+      {
+        timeout: gem2sStepTimeOut,
+        interval: 2000,
+        ...config,
+      }).then((message) => {
+        cy.log('Expecting step to complete and error to be undefined');
+
+        // GEM2S steps doesn't "response" property if it's not error
+        expect(message.response).to.equal(undefined);
+
+        const log = Cypress.log({
+          displayName: 'GEM2S',
+          message: `GEM2S task ${message.taskName} completed - step ${step + 1} of ${numGem2sSteps}`,
+          autoEnd: false,
+        });
+        log.snapshot(`gem2s-step-${step + 1}`);
+        log.end();
+      });
+    });
+  });
+});
+
+Cypress.Commands.add('waitForQc', (experimentId, config = {}) => {
```

Comment:
I don't think there's anything else worthwhile to abstract out. A common mechanism that has been abstracted out is `listenOnWebsocket`. The remaining functionalities depend on the response of GEM2S and QC, which I think should be left unabstracted in the function so that we can easily adapt the behavior to adjust to any changes in the response.